### DISCUSSION TITLE
SPI via Berry?

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#20601] [Q&A] [answered]
I would think spi is available in Berry but i didn't found any infomations.
i would try to build an driver for rfm69

**Comment by s-hadinger** [comment, marked as answer]
No SPI is not available in Berry, mostly because they we hadn't any needs up to now. I generally prefer I2C when chips offer this alternative

**Reply by ChrisVonFrosta** [reply]
Thx, but bad. I would prefer, but this module doesn't offer I2C, and I need this one. 
What options do I have? I have my own C library for 8-bit ATmega. If no built-in option is present, I would build a driver for Tasmota, but I have no entry point on how to do so.
There are several files that have to be updated (`support_tasmota.ino`, `tasmota_template.h`,...), plus adding files like: `xdr_XY_rfm69.ino`, `lib/lib_rf/RFM/RFM69.cpp`?

Could someone give me an overview of what I need to provide? If there is any interest here (and I get it done), I would share it.

or could adding spi support for berry the easier way?
thx

**Comment by barbudor** [comment]
What is the part?
Depending what it is about it could be a xdrv, a xsns or a xnrj
You need to have a look for a driver of the same category using SPI

**Reply by ChrisVonFrosta** [reply]
rfm69 is a radio module (sub-GHz). A german [overview ](https://www.mikrocontroller.net/articles/RFM69)of this module.
Here is the [datasheet](https://www.pollin.de/productdownloads/D810307D.PDF)

i was searching in the repo for files that content "nf24". So i got my assessment from last post (`xdr_XY_rfm69.ino`, `lib/lib_rf/RFM/RFM69.cpp`)

**Reply by Noschvie** [reply]
> rfm69 is a radio module

Nice module...

**Reply by sfromis** [reply]
Here's examples of SPI-based drivers (look for `USE_SPI` in the Tasmota source code)
https://github.com/arendst/Tasmota/blob/development/tasmota/tasmota_xdrv_driver/xdrv_67_mcp23xxx.ino
https://github.com/arendst/Tasmota/blob/development/tasmota/tasmota_xsns_sensor/xsns_80_mfrc522.ino
